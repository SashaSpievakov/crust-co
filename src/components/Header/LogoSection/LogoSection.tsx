import { memo } from 'react';
import { useLocation } from 'react-router-dom';

import mainLogo from '../../../assets/img/main-logo.svg';
import { CustomLink, Heading, Logo, SubHeading } from './LogoSection.styled';

export const LogoSection = memo(() => {
  const location = useLocation();

  return (
    <CustomLink
      to={location.pathname !== '/' ? '/' : '#'}
      tabIndex={location.pathname !== '/' ? 0 : -1}
      ishome={location.pathname === '/' ? 'home' : null}
    >
      <Logo>
        <img width="38" height="54.5" src={mainLogo} alt="Pizza logo" />
        <div>
          <Heading>Crust & Co.</Heading>
          <SubHeading>the most delicious pizzas in town</SubHeading>
        </div>
      </Logo>
    </CustomLink>
  );
});
