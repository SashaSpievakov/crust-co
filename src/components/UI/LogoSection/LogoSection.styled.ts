import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

interface CustomLinkProps {
  ishome: string | null;
}

const CustomLink = styled(Link)<CustomLinkProps>`
  border-radius: 5px;

  &:focus-visible {
    outline: 4px solid ${({ theme }) => theme.colors.secondary};
    outline-offset: 10px;
  }

  ${({ ishome }) =>
    ishome &&
    css`
      cursor: default;

      &:focus-visible {
        outline: none;
      }
    `}
`;

const Logo = styled.section`
  display: flex;

  img {
    margin-right: 15px;
  }

  @media screen and (max-width: 400px) {
    img {
      height: 76px;
    }

    p {
      max-width: 190px;
    }
  }
`;

const Heading = styled.h1`
  color: ${({ theme }) => theme.colors.third};
  font-size: 24px;
  letter-spacing: 1%;
  text-transform: uppercase;
  font-weight: 800;
`;

const SubHeading = styled.p`
  color: ${({ theme }) => theme.colors.grayReverse};

  @media screen and (max-width: 400px) {
    max-width: 190px;
  }
`;

export { CustomLink, Logo, Heading, SubHeading };
