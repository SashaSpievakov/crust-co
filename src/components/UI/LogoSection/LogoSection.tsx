import { memo } from 'react';
import { useLocation } from 'react-router-dom';

import mainLogo from '../../../assets/img/main-logo.svg';
import { CustomLink, Logo, Heading } from './LogoSection.styled';

const LogoSection = memo(() => {
  const location = useLocation();

  return (
    <CustomLink
      to={location.pathname !== '/' ? '/' : '#'}
      tabIndex={location.pathname !== '/' ? 0 : -1}
      ishome={location.pathname === '/' ? 'home' : null}
    >
      <Logo>
        <img width="38" height="54.5" src={mainLogo} alt="Pizza logo" />
        <div>
          <Heading>Pizza Place</Heading>
          <p>the most delicious pizzas in town</p>
        </div>
      </Logo>
    </CustomLink>
  );
});
export default LogoSection;
