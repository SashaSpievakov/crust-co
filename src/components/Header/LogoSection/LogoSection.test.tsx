import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {
  rendererWithProviders,
  renderWithProvidersAndRoutes,
} from '../../../tests/helpers';
import { LogoSection } from './LogoSection';

describe('LogoSection Tests', () => {
  test('renders the LogoSection UI component', () => {
    const snapshot = rendererWithProviders(<LogoSection />);
    expect(snapshot).toMatchSnapshot();
  });

  test('checks link redirect to the Home page', () => {
    renderWithProvidersAndRoutes(<LogoSection />, true, '/cart');
    const mainHeader = screen.getByRole('heading', {
      level: 1,
    });

    userEvent.click(mainHeader);

    const allPizzasHeader = screen.getByRole('heading', {
      level: 2,
    });
    expect(allPizzasHeader).toHaveTextContent(/all pizzas/i);
  });
});
