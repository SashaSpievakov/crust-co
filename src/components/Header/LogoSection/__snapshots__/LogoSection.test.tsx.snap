// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSection Tests renders the LogoSection UI component 1`] = `
Object {
  "children": Array [
    <section
      className="sc-GKYbw eJexPu"
    >
      <img
        alt="Pizza logo"
        height="54.5"
        src="main-logo.svg"
        width="38"
      />
      <div>
        <h1
          className="sc-gsGlKL kJACqs"
        >
          Crust & Co.
        </h1>
        <p
          className="sc-bWOGAC ggShfj"
        >
          the most delicious pizzas in town
        </p>
      </div>
    </section>,
  ],
  "props": Object {
    "className": "sc-fcCDlF ktCmbd",
    "href": "/",
    "ishome": "home",
    "onClick": [Function],
    "tabIndex": -1,
    "target": undefined,
  },
  "type": "a",
}
`;
